restapi/pagination.mdx

```diff
@@ -12,25 +12,134 @@ You can use pagination for the following endpoints:
 - [Run Query](/restapi/endpoints/queryApl)
 - [Run Query (Legacy)](/restapi/endpoints/queryDataset)
 
-## Pagination mechanism
+## Pagination mechanisms
 
-Axiom uses cursor-based pagination for these endpoints. The parameters and mechanisms differ between the current and legacy endpoints.
+You can use one of the following pagination mechanisms:
+
+- Pagination based on timestamp (stable)
+- Pagination based on cursor (preview)
+
+Axiom recommends timestamp-based pagination. Cursor-based pagination is in preview and may result in data loss.
+
+## Timestamp-based pagination
+
+The parameters and mechanisms differ between the current and legacy endpoints.
+
+### Run Query
+
+To use timestamp-based pagination with the Run Query endpoint:
+- Include the [`limit` operator](/apl/tabular-operators/limit-operator) in the APL query of your API request. The argument of this operator determines the number of events to display per page.
+- Use `sort by _time asc` or `sort by _time desc` in the APL query. This returns the results in ascending or descending chronological order. For more information, see [sort operator](/apl/tabular-operators/sort-operator).
+- Specify `startTime` and `endTime` in the body of your API request.
+
+### Run Query (Legacy)
+
+To use timestamp-based pagination with the legacy Run Query endpoint:
+- Add the `limit` parameter to the body of your API request. The value of this parameter determines the number of events to display per page.
```

Comment:
The legacy and tabular APIs allow request based `limit` configuration. This means that the minimum or lesser of the query limit and request limit will be in force up to the maximum internal supported limit that axiom supports server side.

In practice, Axiom will default the limit to some value - currently 1000 events. For pagination of datasets that are greater than 1000 events we've found it best to pin the limit in the request and in the APL query to the same number to avoid the defaults and any confusion w.r.t `min(request_limit, apl_limit, axiom_default_limit, max_axiom_limit)` ...